recipes-extended/xen/files/argo-suppress-debug-messages.patch

```diff
@@ -0,0 +1,58 @@
+################################################################################
+SHORT DESCRIPTION: 
+################################################################################
+Suppress some frequently occurring argo debug messages by default. Only
+print if argo DEBUG is enabled.
+
+################################################################################
+LONG DESCRIPTION: 
+################################################################################
+
+################################################################################
+CHANGELOG 
+################################################################################
+
+################################################################################
+REMOVAL 
+################################################################################
+
+################################################################################
+UPSTREAM PLAN
+################################################################################
+
+################################################################################
+INTERNAL DEPENDENCIES 
+################################################################################
+
+################################################################################
+PATCHES 
+################################################################################
+--- a/xen/common/argo.c
++++ b/xen/common/argo.c
+@@ -1525,7 +1525,7 @@ find_ring_mfns(struct domain *d, struct
+     {
+         ASSERT(ring_info->nmfns == NPAGES_RING(len));
+ 
+-        gprintk(XENLOG_DEBUG, "argo: vm%u ring (vm%u:%x vm%u) %p "
++        argo_dprintk("argo: vm%u ring (vm%u:%x vm%u) %p "
+                 "mfn_mapping %p len %u nmfns %u\n",
+                 d->domain_id, ring_info->id.domain_id,
+                 ring_info->id.aport, ring_info->id.partner_id, ring_info,
+@@ -1739,7 +1739,7 @@ register_ring(struct domain *currd,
+         list_add(&ring_info->node,
+                  &currd->argo->ring_hash[hash_index(&ring_info->id)]);
+ 
+-        gprintk(XENLOG_DEBUG, "argo: vm%u registering ring (vm%u:%x vm%u)\n",
++        argo_dprintk("argo: vm%u registering ring (vm%u:%x vm%u)\n",
+                 currd->domain_id, ring_id.domain_id, ring_id.aport,
+                 ring_id.partner_id);
+     }
+@@ -1779,7 +1779,7 @@ register_ring(struct domain *currd,
+             goto out_unlock2;
+         }
+ 
+-        gprintk(XENLOG_DEBUG,
++        argo_dprintk(
```

Comment:
The two changes above look OK to me and would be welcome as changes submitted to Xen. I'm querying this third one though -- how often are you seeing ring re-registration actually occur?

Reply:
@dozylynx Not often. I grouped it with the others since it appears to fit in the same category, so mostly for consistency's sake.

Reply:
@dozylynx, what do you think? Be consistent and hide the re-registration message as well, or leave that one be and just change the other 2? Thanks

Reply:
Have studied this a bit. There's another log message site in `find_ring_mfns` that will also emit a log message on ring re-registration, in addition to this one. They differ in LOG_LEVEL: this one is `DEBUG` and the other is `INFO`.

Ring re-registration is an interesting operation; it should be rare eg. a guest VM resurrecting connections after resume from some power-saving state. It's also the same code path that would be taken by a ring resize, if that operation were allowed (currently is not but is on the list to enable at some point, with supporting changes).

After consideration, I think both should be switched over to `argo_dprintk`. so: keep this change, and please can a similar one be applied to the ring re-registration printk at the top of `find_ring_mfns`, in this same patch.

Thanks.